src/components/search/data/searchLoader.test.jsx

```diff
@@ -37,17 +39,33 @@ jest.mock('@edx/frontend-platform/config', () => ({
 const mockEnterpriseCustomer = enterpriseCustomerFactory();
 extractEnterpriseCustomer.mockResolvedValue(mockEnterpriseCustomer);
 
+const mockAcademies = [
+  {
+    uuid: 'test-academy-uuid-1',
+  },
+];
+
 const mockQueryClient = {
   ensureQueryData: jest.fn().mockResolvedValue({}),
+  getQueryData: jest.fn().mockResolvedValue(mockAcademies),
 };
 
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: () => ({
+    enterpriseSlug: mockEnterpriseCustomer.slug,
+  }),
+  redirect: jest.fn(),
+}));
```

Comment:
Because we're using `renderWithRouterProvider`, we should not be mocking React Router. The tests are set up to render the `loader` function for a defined route path and the entry in `initialEntries`.

`useParams` is only relevant within a UI component; a React hook can't be rendered within a route loader. As a result, I don't believe this `useParams` mock is actually doing much. The `params` within the `searchLoader` itself are provided by React Router.

As for the `redirect`, instead of mocking `redirect` and asserting it was called, I would recommend not mocking `redirect` and letting React Router actually execute the `redirect` to the expected academy detail page route, asserting that the academy detail route element is rendered instead.